### PR TITLE
v1.1.0

### PR DESCRIPTION
**Pull request by mahmoudElkarargyBS**

### CODE REVIEW

File: tests/heavy-tests/README.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# Heavy Tests Subdirectory
+
+This directory encompasses all the comprehensive tests for the project. It executes all project computations utilizing a
+range of technologies to guarantee the seamless functionality of every aspect.
```
Comment (SAbdulah) [inline tests/heavy-tests/README.md:4]: This directory includes all the heavy tests for the project. It executes all project testing examples using a different set of configurations to guarantee the seamless functionality of every test.

File: USER_MANUAL.md [hunk 2]
```diff
@@ -1,49 +1,53 @@
...
-* To enable tests add **```-t```**
-
-* To enable CUDA add **```-c```**
-
-* To enable MPI add **```-m```**
-
-* To enable Verbose add **```-v```**
 
-* To enable debug mode add **```-d```**
+* To Enable support of HiCMA, add `-H` disabled by default.
```
Comment (SAbdulah) [inline USER_MANUAL.md:22]: to `enable` not `Enable`

File: USER_MANUAL.md [hunk 3]
```diff
@@ -218,45 +224,129 @@ for setting your arguments:
...
+// Generate data by passing your arguments through the configurations, hardware, 
+//and container of the data, which will be filled with the newly generated data.
+ExaGeoStat<double>::ExaGeoStatLoadData(configurations, data);
 ```
 
-##### Real Data 
+#### Using Real Data
 
-- The Data Path must be passed to Configuration.
+Here we use already existing data by providing the path to it:
```
Comment (SAbdulah) [inline USER_MANUAL.md:274]: remove already

File: USER_MANUAL.md [hunk 4]
```diff
@@ -218,45 +224,129 @@ for setting your arguments:
...
+And then using the following code:
+
+```R
+exageostat_data <- simulate_data(kernel=kernel, initial_theta=initial_theta, problem_size=problem_size, dts=dts, dimension=dimension, data_path=data_path)
+
+```
+
+### Location Getters
+
+ExaGeoStat supports locations data of dimension upto 3D, and therefore we have getters for X, Y and Z coordinates.
```
Comment (SAbdulah) [inline USER_MANUAL.md:290]: support spatial locations in 2D and 3D, and therefore...

File: USER_MANUAL.md [hunk 5]
```diff
@@ -218,45 +224,129 @@ for setting your arguments:
...
+```c++
+double *locations_x = exageostat_data->GetLocations()->GetLocationX();
+```
+##### *ExaGeoStat R wrapper*
+```R 
+locations_x <- get_locationsX(data=exageostat_data)
+```
+The subsequent arguments are as follows:
+
+- `exagostat_data`: pointer to ExaGeoStatData object containing the spatial data..
```
Comment (SAbdulah) [inline USER_MANUAL.md:302]: remove one period (.)

File: USER_MANUAL.md [hunk 6]
```diff
@@ -218,45 +224,129 @@ for setting your arguments:
...
-ExaGeoStat<double>::ExaGeoStatLoadData(hardware, configurations, data);
+double *locations_y = exageostat_data->GetLocations()->GetLocationY();
+```
+##### *ExaGeoStat R wrapper*
+```R 
+locations_y <- get_locationsY(data=exageostat_data)
 ```
+The subsequent arguments are as follows:
+
+- `exagostat_data`: pointer to ExaGeoStatData object containing the spatial data..
```
Comment (SAbdulah) [inline USER_MANUAL.md:314]: remove one period

File: man/Data.Rd [hunk 7]
```diff
@@ -0,0 +1,40 @@
+
+% Copyright (c) 2017-2023 King Abdullah University of Science and Technology,
```
Comment (SAbdulah) [inline man/Data.Rd:2]: replace all 2023 with 2024

File: README.md [hunk 8]
```diff
@@ -17,8 +17,7 @@ maintaining traditional practices and embracing contemporary C++ elements like n
 
 
 Vision of ExaGeoStat/ExaGeoStatCPP
-=================
-
+==================================
 The ExaGeoStat/ExaGeoStatCPP project is a collaboration between the KAUST Spatial Statistics group and the Extreme Computing Research
 Center (ECRC). Its contribution lies not in a new algorithm nor a new dataset,
```
Comment (SAbdulah) [inline README.md:22]: Please remove "its contribution...

File: README.md [hunk 9]
```diff
@@ -55,18 +54,27 @@ numerical accuracy. This further expands practical problem sizes for
 statisticians with modest computational resources.
 
 
+ExaGeoStatR : R wrapper of ExaGeoStat
+=====================================
+ R is a powerful and versatile tool for scientific computing, offering a wide range of statistical and graphical
+ techniques, strong community support, and the flexibility to integrate with other programming languages. 
+ Its open-source nature and extensive package ecosystem make it an invaluable resource for researchers and data scientists.
+
```
Comment (SAbdulah) [inline README.md:62]: Interface not wrapper

File: CHANGELOG.md [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+- Modified the calculation of P for all kernels.
+- Adjusted CMake variables.
+- Revised the process of finding BLASPP and Catch2 libraries.
+- Updated doxygen documentation.
+- Split the synthetic generator functions into BitHelper class and Locations generator class.
+- Created a Bassel Function helper for kernels.
+- Cleaned the code base for better readability.
+
+### Removed
+- Eliminated non-stationary kernel support.
```
Comment (SAbdulah) [inline CHANGELOG.md:35]: The non-stationary kernel support.

File: CHANGELOG.md [hunk 11]
```diff
@@ -0,0 +1,40 @@
...
+- Removed Find OpenMP, LAPACKPP, and CuSOLVER.
```
Comment (SAbdulah) [inline CHANGELOG.md:36]: I cannot understand what was removed?

File: CMakeLists.txt [hunk 12]
```diff
@@ -67,86 +73,100 @@ else ()
...
-# Print installation path of Exageostat.
-message(STATUS "Installation path : ${EXAGEOSTAT_INSTALL_PREFIX}")
+# Add all dependencies for ExaGeoStatCPP
+#-----------------------------
+
+# Print installation path of ExaGeoStatCPP.
+message(STATUS "Installation path : ${CMAKE_INSTALL_PREFIX}")
 
-# EXAGEOSTAT depends on a Hwloc
+# ExaGeoStatCPP depends on a HWLoc
```
Comment (SAbdulah) [inline CMakeLists.txt:105]: remove `a`

File: CMakeLists.txt [hunk 13]
```diff
@@ -67,86 +73,100 @@ else ()
...
 # -------------------------------
 include(ImportHwloc)
 list(APPEND STARPU_COMPONENT_LIST "HWLOC")
-
 string(REPLACE ";" " " STARPU_COMPONENT_STRING "${STARPU_COMPONENT_LIST}")
 
-# EXAGEOSTAT depends on a runtime
+# ExaGeoStatCPP depends on a StarPu runtime
```
Comment (SAbdulah) [inline CMakeLists.txt:111]: remove `a`. StarPU not StarPu

File: README.md [hunk 14]
```diff
@@ -54,108 +53,93 @@ This low-rank matrix approximation permits the exploitation of the data sparsity
...
+   install.packages("assert")
+   ```
+
+2. close the R prompt and return to the terminal. Run the following command, make sure your current path is the ExaGeoStat project directory
+
+   ```commandline
+   R CMD INSTALL . --configure-args="-r"
+   ```
+
+> For more detailed information on setting up ExaGeoStat with different configurations and enabling technologies such as CUDA, MPI, R, etc., please refer to the [User Manual](USER_MANUAL.md)
```
Comment (SAbdulah) [inline README.md:104]: installing not setting up

File: USER_MANUAL.md [hunk 15]
```diff
@@ -1,83 +1,104 @@
...
+6. [Supported operations](#supported-operations)
+7. [Manuals](#Manuals)
+8. [Contributing](#contributing)
+
+## ExaGeoStatCPP
+> Current Version of ExaGeoStatCPP: 1.1.0
+### Supported Operations:
+1. (Data Generation): Generating large geospatial synthetic datasets using  dense, Diagonal Super-Tile (DST) and Tile Low-Rank (TLR) approximation techniques.
+2. (Data Modeling): Modeling large geospatial datasets on dense, Diagonal Super-Tile (DST) and Tile Low-Rank (TLR) approximation techniques through the Maximum likelihood Estimation (MLE) operation.
+3. (Data Prediction): Predicting missing measurements on given locations using dense, Diagonal Super-Tile (DST), and Tile Low-Rank (TLR) approximation techniques.
```
Comment (SAbdulah) [inline USER_MANUAL.md:19]: Let's use just the abbreviations after introducing them once: DST and TLR.

File: CHANGELOG.md (hunk 11 above)
Comment (mahmoudElkarargyBS) [reply CHANGELOG.md:36]: Find CMake files

Comment (mahmoudElkarargyBS) [reply CHANGELOG.md:36]: I update it to - Removed FindOpenMP.cmake, FindLAPACKPP.cmake, and FindCuSOLVER.cmake.

File: USER_MANUAL.md [hunk 16]
```diff
@@ -218,45 +240,128 @@ for setting your arguments:
     synthetic_data_configurations.SetKernelName("BivariateSpacetimeMaternStationary");
     synthetic_data_configurations.SetPrecision(exageostat::common::double);
 ```
-### Initialize the Hardware
+### Initialize and Finalize the Hardware
```
Comment (SAbdulah) [inline USER_MANUAL.md:243]: Are you mean hardware struct? data structure?

**Pull request merged**